.github/workflows/changelog-existence.yml

```diff
@@ -0,0 +1,106 @@
+# **what?**
+# Checks that a file has been committed under the /.changes directory
+# as a new CHANGELOG entry.  Cannot check for a specific filename as
+# it is dynamically generated by change type and timestamp.
+#
+
+# **why?**
+# Ensure code change gets reflected in the CHANGELOG.
+
+# **when?**
+# This will run when called in a workflow.
+
+# Example Usage including required permissions
+
+# permissions:
+#   contents: read
+#   pull-requests: write
+#
+# name: Check Changelog Entry
+# jobs:
+#   changelog:
+#     uses: dbt-labs/actions/.github/workflows/changelog-check.yml@main
+#    with:
+#      changelog_comment: 'Thank you for your pull request! We could not find a changelog entry for this change. For details on how to document a change, see [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md#adding-changelog-entry).'
+#      skip_label: 'Skip Changelog'
+#    secrets: inherit # this is only acceptable because we own the action we're calling
+#
+
+name: Check Changelog Entry Exists
+
+on:
+  workflow_call:
+    inputs:
+      changelog_comment:
+        description: The comment text when no changelog is found
+        type: string
+        required: true
+      skip_label:
+        description: Label on the PR that indicates CHANGELOGs are not required.
+        type: string
+        required: true
+
+jobs:
```

Comment:
Should this be a composite action instead of a workflow? Can you show an example how it would be used, or open a PR in another repo showing that?